lib/msf/core/modules/external/templates/evasion.erb

```diff
@@ -0,0 +1,49 @@
+require 'msf/core/modules/external/bridge'
+require 'msf/core/module/external'
+
+class MetasploitModule < Msf::Evasion
+  include Msf::Module::External
+
+  def initialize
+    super({
+    <%= common_metadata meta %>
+      'References'      =>
+        [
+          <%= meta[:references] %>
+        ],
+      'Platform'        => [<%= meta[:platform] %>],
+      'Arch'            => [<%= meta[:arch] %>],
+      'Targets'         =>
+        [
+          <%= meta[:targets] %>
+        ],
+      'DefaultOptions' =>
+        {
+          <%= meta[:default_options] %>
+        }
+      })
+
+      register_options([
+        <%= meta[:options] %>
+      ])
+
+      register_advanced_options([
+        <%= meta[:advanced_options] %>
+      ])
+  end
+
+  def run
+    args = datastore.to_h.merge(
+      # XXX: JSON-RPC requires UTF-8, so we Base64-encode the binary payload
+      payload_encoded: Rex::Text.encode_base64(payload.encoded),
+      payload_raw: Rex::Text.encode_base64(payload.raw),
+      target: {
```

Comment:
How would an option like this map to the command line (https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/modules/external/python/metasploit/cli.py or https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/modules/external/cli.rb)?

Reply:
Not sure I understand the question. This use case was I need to be able to extract the architecture corresponding to the target the user had selected so I could use it while processing the payload.

If the target needed to be specified from the command line, I'd imagine it would best be done by name and then a target object would make it's way here as if it would when the module is run from the framework. From my experience this matches the use of target by some modules to perform certain context-sensitive actions which I guess externals haven't been doing yet? If this needed to be removed, is there a way that the target name is passed to the external module that I could use to recover this information at run time?

Reply:
I think just expecting the name to be passed in and matching that back up to the rest of the information during the module run would be the best option. You can still set option here, but it should be simple enough to be settable when run manually.